File: algorithm/53. Maximum Subarray/53.cpp
public:
    int maxSubArray(vector<int>& nums) {
        int sum=nums[nums.size()-1];
        int maxSum= sum;
        for (int i = nums.size()-2;i>=0;i--)
        {
            sum = max(nums[i], sum + nums[i]);
            maxSum = max(maxSum, sum);
        }
        return maxSum;
    }
};

File: algorithm/53. Maximum Subarray/53.java
class Solution {
    public int maxSubArray(int[] nums) {
        if( nums.length == 0)   return 0;
        int sum = nums[0];
        int maxsum = sum;
        for(int i = 1; i < nums.length; i++)
        {
            sum = Math.max(nums[i], nums[i] + sum);
            maxsum = Math.max(maxsum, sum);
        }
        return maxsum;
    }
}
